### PR TITLE
feat: add keep3r job again

### PR DESCRIPTION
**Pull request by 0xsambugs**
We are now re-adding the keep3r job that executes swaps

Note: this PR only adds the constructor for the contract

### CODE REVIEW

File: contracts/DCAKeep3rJob/DCAKeep3rJob.sol [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    // We are setting the super admin role as its own admin so we can transfer it
+    _setRoleAdmin(SUPER_ADMIN_ROLE, SUPER_ADMIN_ROLE);
+    _setRoleAdmin(CAN_SIGN_ROLE, SUPER_ADMIN_ROLE);
+    _setupRole(SUPER_ADMIN_ROLE, _superAdmin);
+
+    for (uint256 i = 0; i < _initialCanSign.length; i++) {
+      _setupRole(CAN_SIGN_ROLE, _initialCanSign[i]);
+    }
+  }
```
Comment (0xged) [inline contracts/DCAKeep3rJob/DCAKeep3rJob.sol:43]:
```suggestion
    for (uint256 i = 0; i < _initialCanSign.length;) {
      _setupRole(CAN_SIGN_ROLE, _initialCanSign[i]);
      unchecked {
        i++;
      }
    }
  }
```

Ethereum mainnet deployment are costly sir 😋

File: contracts/interfaces/IDCAKeep3rJob.sol [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+pragma solidity >=0.8.7 <0.9.0;
+
+import 'keep3r-v2/solidity/interfaces/IKeep3r.sol';
+
+interface IDCAKeep3rJob {
+  /// @notice A struct that contains the swapper and nonce to use
+  struct SwapperAndNonce {
+    address swapper;
+    uint96 nonce;
+  }
```
Comment (0xged) [inline contracts/interfaces/IDCAKeep3rJob.sol:11]: Would it make sense to add a `@dev` tag indicating max nonce? Or is it just too obvious?

File: package.json [hunk 3]
```diff
@@ -64,7 +64,8 @@
     "@mean-finance/dca-v2-core": "3.1.0",
     "@mean-finance/deterministic-factory": "1.6.0",
     "@mean-finance/swappers": "1.1.0",
-    "@openzeppelin/contracts": "4.7.3"
+    "@openzeppelin/contracts": "4.8.0",
+    "keep3r-v2": "^1.0.0"
```
Comment (0xged) [inline package.json:68]:
```suggestion
    "keep3r-v2": "1.0.0"
```

**Review by 0xged — APPROVED** [review]
Looks good!

File: contracts/interfaces/IDCAKeep3rJob.sol (hunk 2 above)
Comment (0xsambugs) [reply contracts/interfaces/IDCAKeep3rJob.sol:11]: I would say it's unnecessary

**Pull request merged**